src/gen_mod.erl

```diff
@@ -277,27 +282,21 @@ config_spec(Module) ->
 -spec wait_for_process(atom() | pid() | {atom(), atom()}) -> 'ok'.
 wait_for_process(Process) ->
     MonitorReference = erlang:monitor(process, Process),
-    wait_for_stop(Process, MonitorReference).
-
-
--spec wait_for_stop(atom() | pid() | {atom(), atom()}, reference()) -> 'ok'.
-wait_for_stop(Process, MonitorReference) ->
-    receive
-        {'DOWN', MonitorReference, _Type, _Object, _Info} ->
-            ok
-    after 5000 ->
+    case wait_for_stop(MonitorReference) of
+        ok -> ok;
+        timeout ->
             catch exit(whereis(Process), kill),
-            wait_for_stop1(MonitorReference)
+            wait_for_stop(MonitorReference),
```

Comment:
This is a significant change of behaviour - before it waited up to 2*5 = 10 seconds, now it can get stuck in an infinite loop. Of course it should't if the `DOWN` message is received, but I wouldn't set up such a trap in the code. Let's have a maximum retry count here.

Reply:
It's not a recursive call, `wait_for_process/1` function just calls `wait_for_stop/1` 2 times. So the behaviour is exactly the same.